metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1244,56 +1245,77 @@ def visit_join_to_time_spine_node(self, node: JoinToTimeSpineNode) -> SqlDataSet
         parent_data_set = node.parent_node.accept(self)
         parent_alias = self._next_unique_table_alias()
 
-        # Build time spine dataset
-        metric_time_dimension_instance: Optional[TimeDimensionInstance] = None
-        for instance in parent_data_set.metric_time_dimension_instances:
-            if len(instance.spec.entity_links) == 0:
-                # Use the instance with the lowest granularity
-                if not metric_time_dimension_instance or (
-                    instance.spec.time_granularity < metric_time_dimension_instance.spec.time_granularity
-                ):
-                    metric_time_dimension_instance = instance
         assert (
-            metric_time_dimension_instance
-        ), "Can't join to time spine without metric time. Validations should have prevented this."
-        metric_time_dimension_column_name = self.column_association_resolver.resolve_spec(
-            metric_time_dimension_instance.spec
+            len(node.requested_agg_time_dimension_specs) > 0
+        ), "Must have at least one value in requested_agg_time_dimension_specs for JoinToTimeSpineNode."
+
+        # Determine if the time spine join should use metric_time or the agg_time_dimension (metric_time takes priority).
+        agg_time_dimension_for_join = node.requested_agg_time_dimension_specs[0]
+        for spec in node.requested_agg_time_dimension_specs[1:]:
+            if spec.element_name == METRIC_TIME_ELEMENT_NAME:
```

Comment:
This is no longer guaranteed to be the lowest granularity, right? Is that intentional? It seems like it doesn't matter, but it's not clear at this point if that's really the case or not.

Reply:
The granularity check happens a few lines down!